jvm-packages/xgboost4j-spark-gpu/src/main/scala/ml/dmlc/xgboost4j/scala/spark/rapids/GpuXGBoost.scala

```diff
@@ -0,0 +1,581 @@
+/*
+ Copyright (c) 2014 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.scala.spark.rapids
+
+import scala.collection.JavaConverters._
+
+import ml.dmlc.xgboost4j.gpu.java.{CudfColumn, CudfColumnBatch}
+import ml.dmlc.xgboost4j.java.spark.GpuColumnBatch
+import ml.dmlc.xgboost4j.java.{Rabit, XGBoostError}
+import ml.dmlc.xgboost4j.scala.{Booster, DMatrix, EvalTrait, ExternalCheckpointManager, ObjectiveTrait}
+import ml.dmlc.xgboost4j.scala.{XGBoost => SXGBoost, _}
+import ml.dmlc.xgboost4j.scala.spark._
+import ml.dmlc.xgboost4j.scala.spark.params.BoosterParams
+import org.apache.commons.logging.LogFactory
+import org.apache.hadoop.fs.FileSystem
+
+import org.apache.spark.sql.catalyst.encoders.RowEncoder
+import org.apache.spark.sql.{DataFrame, Dataset, Row}
+import org.apache.spark.sql.functions._
+import org.apache.spark.{GpuParallelismTracker, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.util.LongAccumulator
+
+private[spark] object GpuXGBoost {
```

Comment:
Why do we need a new fit function?

Reply:
this new fit is an interface called by XGBoostClassifier/XGBoostRegressor